servicetalk-http-api/src/main/java/io/servicetalk/http/api/BlockingStreamingToStreamingService.java

```diff
@@ -142,10 +136,16 @@ public void cancel() {
                     tiCancellable.setDone(cause);
                     if (response == null || response.markMetaSent()) {
                         safeOnError(subscriber, cause);
-                    } else if (payloadWriterOuter.markSubscriberComplete()) {
-                        payloadProcessor.onError(cause);
                     } else {
-                        LOGGER.error("An exception occurred after the response was sent", cause);
+                        try {
+                            payloadProcessor.onError(cause);
+                        } finally {
+                            try {
+                                payloadWriter.close(cause);
```

Comment:
Discussed offline: we should adjust `payloadProcessor` to correctly propagate the error instead of using `payloadWriter.close(cause)` in this case.

Reply:
I think we can use both, to make sure the `payloadWriter` resources are reclaimed.

Reply:
The larger changes we discussed related to the `CompletableProcessor` will have to be delayed until my followup PR which fixes payload body and trailers propagation (otherwise gRPC `ErrorHandlerTest` fails for blocking streaming use cases).